manifests/adjoin/password.pp

```diff
@@ -8,18 +8,26 @@
   $_user     = $::centrify::join_user
   $_password = $::centrify::join_password
   $_domain   = $::centrify::domain
+  $_zone     = $::centrify::zone
 
-  exec { 'adjoin_with_password':
-    path        => '/usr/bin:/usr/sbin:/bin',
-    command     => "adjoin -w -u ${_user} -p ${_password} ${_domain}",
-    unless      => "adinfo -d | grep ${_domain}",
-    refreshonly => true,
-  }->
+  if $_zone!=undef{
```

Comment:
I think this could be made a little cleaner by just testing for $_zone and setting a variable that contains the adjoin command. Then only one exec would be needed. For example:
```
if $_zone {
  $_adjoin_command = <zone join>
}
else {
  $_adjoin_command = <auto zone join>
}
```

Could you also extend zone support to keytab joins for consistency?